frontend/scripts/gameLogicInterface.js

```diff
@@ -191,11 +191,27 @@ export default function createGame(initialGameState = emptyGameState) {
 		moveLeft: function() {
 
 		},
+          
+
+        moveDown: function() {
+			let { x, y } = this.gameState.activeTetromino.position;
+			
+			this.gameState.activeTetromino.position = { x: x, y: y - 1 };
+		if (!this.isStateValid(this.gameState.activeTetromino)) {
+			this.gameState.activeTetromino.position = { x: x, y: y };
+			return false;

```

Comment:
It's a little strange to set the `activeTetromino` to the next position, to then check whether it's valid, and then potentially changing it back to what it was before. Can you think about how you could rewrite this logic to not have to do this?

Also, the `moveDown` function is not expecting anything to be returned